### PR TITLE
[Button] Remove Children Prop

### PR DESCRIPTION
**Pull request by emilyrohrbough**
### Summary
After much discussion, remove children props in v2 release until a use-case and style is presented.

- Remove `children` prop
- Fix site examples

### CODE REVIEW

File: packages/terra-site/src/examples/alert/ActionAndDismissibleExample.jsx [hunk 1]
```diff
@@ -32,7 +32,7 @@ class ActionAndDismissibleExample extends React.Component {
     let alertElem = '';
     if (!this.state.isDismissed) {
       alertElem = (
-        <Alert type={Alert.Opts.Types.CUSTOM} onDismiss={this.handleDismiss} customStatusColor="orange" action={<Button text="Action" size="medium" variant="primary" onClick={this.actionFunc} />} >
+        <Alert type={Alert.Opts.Types.CUSTOM} onDismiss={this.handleDismiss} customStatusColor="orange" action={<Button text="Action" size="medium" variant="emphasis" onClick={this.actionFunc} />} >
```
Comment (bjankord) [inline packages/terra-site/src/examples/alert/ActionAndDismissibleExample.jsx]: If you are using Button v2, it won't have the size prop anymore right?

File: packages/terra-site/src/examples/alert/ActionExample.jsx [hunk 2]
```diff
@@ -35,7 +35,7 @@ class ActionExample extends React.Component {
           </code>
         </div>
         <br />
-        <Alert type={Alert.Opts.Types.WARNING} action={<Button text="Action" size="medium" variant="primary" onClick={this.actionFunc} />} >
+        <Alert type={Alert.Opts.Types.WARNING} action={<Button text="Action" size="medium" variant="emphasis" onClick={this.actionFunc} />} >
```
Comment (bjankord) [inline packages/terra-site/src/examples/alert/ActionExample.jsx]: If you are using Button v2, it won't have the size prop anymore right?

Comment (emilyrohrbough) [reply packages/terra-site/src/examples/alert/ActionExample.jsx]: Oops. No it won't.

**Comment by bjankord** [comment]
Travis CI build is [failing out due to time limit](https://api.travis-ci.org/v3/job/325175710/log.txt). `The job exceeded the maximum time limit for jobs, and has been terminated.`

**Pull request merged**